LOCAL_DEV.md

```diff
@@ -0,0 +1,234 @@
+REACT_APP_CONTRACT_NON_FUNGIBLE_TOKEN=
+
+REACT_APP_API_KIBBLE_MINT=http://localhost:3000/v1/kibbles/mint
+REACT_APP_API_KITTY_ITEM_MINT=http://localhost:3000/v1/kitty-items/mint
+REACT_APP_API_MARKET_ITEMS_LIST=http://localhost:3000/v1/market/latest
+
+REACT_APP_CONTRACT_KIBBLE=
+REACT_APP_CONTRACT_KITTY_ITEMS=
+REACT_APP_CONTRACT_KITTY_ITEMS_MARKET=
+```
+
+Here you can see we're setting up to connect to the Flow Emulator and the `fcl-dev-wallet` by specifying their
+respective host interfaces.
+
+In the next step we'll add the missing values in our `.env.local` files for: 
+- The Flow account that we'll deploy the Kitty-Items contracts to.
+- The private key for that account.
+- The addresses of the deployed contracts we'll use for this project
+
+### 5. Start the Flow emulator
+
+To find the values we'll need to complete settin up our environment, we'll need to start the Flow emulator.
+
+```sh
+flow emulator --persist
+```
+
+We're including the persist option, which will allow us to maintin the state of the emulatoed blockchain when we stop and restart the emulator. 
+You should see the following output in your terminal:
+
+```sh
+INFO[0000] ⚙️   Using service account 0xf8d6e0586b0a20c7  serviceAddress=f8d6e0586b0a20c7 serviceHashAlgo=SHA3_256 servicePrivKey=f8e188e8af0b8b414be59c4a1a15cc666c898fb34d94156e9b51e18bfde754a5 servicePubKey=6e70492cb4ec2a6013e916114bc8bf6496f3335562f315e18b085c19da659bdfd88979a5904ae8bd9b4fd52a07fc759bad9551c04f289210784e7b08980516d2 serviceSigAlgo=ECDSA_P256
+INFO[0000] 📜  Flow contracts                             FlowFees=0xe5a8b7f23e8b548f FlowServiceAccount=0xf8d6e0586b0a20c7 FlowStorageFees=0xf8d6e0586b0a20c7 FlowToken=0x0ae53cb6e3f42a79 FungibleToken=0xee82856bf20e2aa6
+INFO[0000] 🌱  Starting gRPC server on port 3569          port=3569
+INFO[0000] 🌱  Starting HTTP server on port 8080          port=8080
+```
+This output contains the addresses for the contracts that are built into the emulator, as well as the address of the `FlowServiceAccount`, a pre-generated account you can use to easily get started deploying code to the emulator. We'll use the following values from the emulator's output in our `.env.local` files: 
+
+(☝️ **Note**: your local emulator's output may be different. Be sure to use values from _your own shell_ to ensure they match with your local environment).
+
+#### Add the Flow Service Account address and private key
+
+In `web/.env.local` add the value of `FlowServiceAccount` in your emulator's output as:
+  - `REACT_APP_CONTRACT_KIBBLE`
+  - `REACT_APP_CONTRACT_KITTY_ITEMS`
+  - `REACT_APP_CONTRACT_KITTY_ITEMS_MARKET`
+
+In `api/.env.local`: 
+  - Add the value of `FlowServiceAccount` in your emulator's output as:
+    - `MINTER_ADDRESS`
+
+  - Add the value of `servicePrivKey` from your emulator's output as:
+    - `MINTER_PRIVATE_KEY`
+
+#### Add the Fungible Token contract address
+
+Next we'll update the address of the emeulator's `FungibleToken` contract. 
```

Comment:
The emulator should be started using CLI with command `flow emulator --persist` this way it will use `flow.json` as a configuration file and hence the service account address, private key, and the public key will be deterministic which will allow you to potentially remove all these value assignments from the emulator output. All the contracts that are already included in the emulator will also be always on the same address so you can also remove that.

Reply:
Yes, but I again, I wanted to be explicit so people understand where certain values are coming from (because they have to set them themselves)
 
It might be worth having more instructions about how to use the emulator with `flow.json` file. (I have a task to do that for this sprint actually) and document the `--persist` option there.

Reply:
I really think we should be encouraging users to use flow cli to start tools like the emulator and later wallet. Wasn't that our plan to make it easier since it can be bundled. If not running the emulator using flow.json you are risking that the values you just set in those env variables won't be the same next time you restart the laptop.

Reply:
Hmm, ok I'll revise this portion. I'll definitely mention the link between the cli and `flow.json` but im inclined to leave the manual step of adding the address beaxuse I thnik it;s important to introduce `flow.json`. I think getting people to touch it will force them to think about it, ask questions ...etc. Instead of avoiding it or treating it like a black box/magic. (So, i think it;s a benefit to manually edit, even though you don't _have_ to)

Reply:
I'll admit that I was trying to increase the difficulty, _a little bit_. 
I think, the more difficult the obstacle you overcome, the more of an impact that experience has long-term. 
We're not going to scare anyone away, because they don't know any better. _We_ get to show them around. Folks like you will know they can write the command differently, but I bet you learned the long command first?

Reply:
@srinjoy

Reply:
It's your call for the approach, just be careful so there is not misconfiguration of CLI and emulator as it will cause CLI to not sign correctly. Also, my effort has been to prefer using CLI `config` command to edit flow.json as it offers validation and it abstracts format changes. Manualy editing flow.json had many reports and issues on discord because developers make mistakes. It's just things to keep in mind if you plan to go this route.

Reply:
Ah nice! I wasn't aware you could change the config from the cli 😅
This is definitely the way to go.

Reply:
https://github.com/onflow/flow-cli/blob/master/docs/manage-configuration.md 
Speaking of which I believe docs have to be rebuilt for this to be visible.